src/testing/burgers_limiter.cpp

```diff
@@ -0,0 +1,220 @@
+            grid->add_periodicity(matched_pairs);
+#else
+            std::vector<dealii::GridTools::PeriodicFacePair<typename dealii::parallel::distributed::Triangulation<PHILIP_DIM>::cell_iterator> > matched_pairs;
+            dealii::GridTools::collect_periodic_faces(*grid, 0, 1, 0, matched_pairs);
+            if (dim >= 2) dealii::GridTools::collect_periodic_faces(*grid, 2, 3, 1, matched_pairs);
+            if (dim >= 3) dealii::GridTools::collect_periodic_faces(*grid, 4, 5, 2, matched_pairs);
+            grid->add_periodicity(matched_pairs);
+#endif
+           grid->refine_global(igrid);
+           // grid->refine_global(log2(17)); //make sure shock is not on the face
+            pcout << "Grid generated and refined" << std::endl;
+
+            const unsigned int n_global_active_cells2 = grid->n_global_active_cells();
+            double n_dofs_cfl = pow(n_global_active_cells2, dim) * pow(poly_degree + 1.0, dim);
+            double delta_x = (PHILIP_DIM == 2) ? (right - left) / pow(n_global_active_cells2, (1.0/dim)) : (right - left) / pow(n_dofs_cfl, (1.0 / dim));
+            all_parameters_new.ode_solver_param.initial_time_step = (PHILIP_DIM == 2) ? (1.0 / 14.0) * pow(delta_x, 1.0/*(4.0/3.0)*/) : (1.0 / 24.0) * pow(delta_x, (1.0));
+            pcout << "time_step:   " << all_parameters_new.ode_solver_param.initial_time_step << std::endl;
+
+            //allocate dg
+            std::shared_ptr < PHiLiP::DGBase<dim, double> > dg = PHiLiP::DGFactory<dim, double>::create_discontinuous_galerkin(&all_parameters_new, poly_degree, poly_degree, grid_degree, grid);
+            pcout << "dg created" << std::endl;
+            dg->allocate_system();
+
+            //initialize IC
+            pcout << "Setting up Initial Condition" << std::endl;
+            //Create initial condition function
+            std::shared_ptr< InitialConditionFunction<dim, nstate, double> > initial_condition_function =
+                InitialConditionFactory<dim, nstate, double>::create_InitialConditionFunction(&all_parameters_new);
+            SetInitialCondition<dim, nstate, double>::set_initial_condition(initial_condition_function, dg, &all_parameters_new);
+
+            //get the global max and min of initial condition.
+            //dg->get_global_max_and_min_of_solution();
+            //dg->h = delta_x;
+            // Create ODE solver using the factory and providing the DG object
+            std::shared_ptr<ODE::ODESolverBase<dim, double>> ode_solver = ODE::ODESolverFactory<dim, double>::create_ODESolver(dg);
+
+            if (!all_parameters_new.limiter_param.use_OOA)
+            {
+                double finalTime = 2.0;
+                ode_solver->advance_solution_time(finalTime);
+            }
+            else
+            {
+                double finalTime = (PHILIP_DIM == 2) ? 0.05 : 0.15;//Comparison with 2010 Zhang Shu Paper
+
+                ode_solver->current_iteration = 0;
+
```

Comment:
similar comment about using flow_solver

Reply:
Created flow_solver case to simply everything!